src/Engine/Graphics/RenderBase.cpp

```diff
@@ -245,6 +255,165 @@ void RenderBase::DrawSpriteObjects() {
     }
 }
 
+void RenderBase::PrepareDecorationsRenderList_ODM() {
```

Comment:
as with ODM_ProcessPartyActions, i feel like it was better off as a standalone thing
think of it as an added complexity layer over both render and the game world state (glue logic)
render class is already pretty bloated, and half of logic in this  method doesn't even relate to the renderer...

the perfect endstate for this would be,  "game world updater/manager" (operates with objects i.e decorations) -> calls into -> "visibility detection (frustrum part)" -> gets list of visible objects -> does some additional logic (seasons etc, probably best if a separate method for that too)  -> for each remaining decoration, passes corresponding "sprites" -> "renderer"

Reply:
This needs moving again true. At this stage I was trying to eliminate unnecessary duplicates of functions between GL/ D3D renderers.